localstack-core/localstack/services/lambda_/event_source_mapping/pollers/sqs_poller.py

```diff
@@ -216,3 +216,14 @@ def get_queue_url(queue_arn: str) -> str:
     account_id = parsed_arn["account"]
     name = parsed_arn["resource"]
     return f"{host}/{account_id}/{name}"
+
+
+def message_attributes_to_lower(message_attrs):
+    """Convert message attribute details (first characters) to lower case (e.g., stringValue, dataType)."""
+    message_attrs = message_attrs or {}
+    for _, attr in message_attrs.items():
+        if not isinstance(attr, dict):
+            continue
+        for key, value in dict(attr).items():
+            attr[first_char_to_lower(key)] = attr.pop(key)
```

Comment:
Is it possible that we have nested dicts whose casing we want to change?

If so, we could do something like (havent checked this properly):
```python3
        for key, value in dict(attr).items():
            attr[first_char_to_lower(key)] = message_attributes_to_lower(attr.pop(key))
```

Reply:
good test case to add :)